File: 9613 - GCD 합/9613.cpp
#include <bits/stdc++.h>
using namespace std;

int gcd(int a, int b){
    return b ? gcd(b, a % b) : a;
}

int main(){
    cin.tie(0); ios::sync_with_stdio(0);
    
    int t; cin >> t;
    
    for (int i = 0; i < t; i++){
        long long sum = 0;
        int n; cin >> n;
        int arr[100] = {};
        for (int j = 0; j < n; j++) 
            cin >> arr[j];

        for (int j = 0; j < n - 1; j++){
            for (int k = j + 1; k < n; k++){
                sum += gcd(max(arr[j], arr[k]), min(arr[j], arr[k]));
            }
        }
        cout << sum << '\n';
    }
}

File: 9613 - GCD 합/9613.py
import sys

def gcd(a, b):
    while True:
        c = a % b
        if c == 0: return b
        a = b
        b = c

for i in range(int(sys.stdin.readline())):
    lst = list(map(int, sys.stdin.readline().split()))
    Sum = 0;
    for j in range(1, lst[0]):
        for k in range(j + 1, lst[0] + 1):
            Sum += gcd(max(lst[j], lst[k]), min(lst[j], lst[k]))
    print(Sum)
